src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -596,6 +596,11 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
     onlyOneBlock: boolean = false
   ) => {
     const nextBlock = this.#readyNextBlock(this.blocks.latest, timestamp);
+
+    // if block time is incremental, adjustments should only apply once, otherwise they accumulate with each block.
+    if (this.#options.miner.timestampIncrement !== "clock") {
```

Comment:
I wonder if this would make more sense to do in the `#adjustedTime` function, right after we use it. 
From:
```TypeScript
#adjustedTime = (precedingTimestamp: Quantity) => {
  const timeAdjustment = this.#timeAdjustment;
  const timestampIncrement = this.#options.miner.timestampIncrement;
  if (timestampIncrement === "clock") {
    return Math.floor((Date.now() + timeAdjustment) / 1000);
  } else {
    return (
      precedingTimestamp.toNumber() +
      Math.floor(timeAdjustment / 1000) +
      timestampIncrement.toNumber()
    );
  }
};
```
to:
```TypeScript
#adjustedTime = (precedingTimestamp: Quantity) => {
  let timeAdjustment = this.#timeAdjustment;
  const timestampIncrement = this.#options.miner.timestampIncrement;
  if (timestampIncrement === "clock") {
    return Math.floor((Date.now() + timeAdjustment) / 1000);
  } else {
    const time =
      precedingTimestamp.toNumber() +
      Math.floor(timeAdjustment / 1000) +
      timestampIncrement.toNumber();
    timeAdjustment = 0;
    return time;
  }
};
```

Just so we're keeping all of this time logic in one area rather than randomly having it in this mine block.

Reply:
I didn't put it in there, because sometimes we call `#adjustedTime`, but don't persist the block - for instance the `pending` block changes that you've got in PR. 

I agree that it's not _super_ nice to have it thrown in the middle of here, but it needs to only happen if the block's being persisted for realsies.

I would like to tidy this up in a rewrite of the block-time stuff, but it's been stressing me out, so I've put it on the backburner :P

Reply:
That makes sense! I'm very much looking forward to your time management refactor - there seems to be a lot of tech-debt wrapped up in how we manage time. But let's get these bugs fixed now if we can!